server/src/main/java/org/elasticsearch/action/resync/TransportResyncReplicationAction.java

```diff
@@ -135,6 +135,7 @@ protected WriteReplicaResult shardOperationOnReplica(ResyncReplicationRequest re
                 }
             }
         }
+        replica.trimTranslog(request.getBelowTermId(), request.getTrimmedAboveSeqNo());
```

Comment:
I think we need a condition on the fact that `getTrimmedAboveSeqNo` is set? This smells like a problem with test coverage too.